eureka2-testkit/src/test/java/com/netflix/eureka2/testkit/embedded/server/EmbeddedWriteServerTest.java

```diff
@@ -32,7 +34,9 @@ public void testRegistrationAndDiscoveryServices() throws Exception {
                 .build();
 
         InstanceInfo instanceInfo = SampleInstanceInfo.DiscoveryServer.build();
-        eurekaClient.register(instanceInfo).toBlocking().lastOrDefault(null);
+        RegistrationRequest request = eurekaClient.connect(Observable.just(instanceInfo));
+        request.subscribe();
+        request.getInitObservable().toBlocking().lastOrDefault(null);
```

Comment:
The caller need to subscribe() to the request to initiate the registration, so it makes more sense for this to be top layer observable.